src/Tempest/Support/tests/ArrayHelperTest.php

```diff
@@ -1381,4 +1381,190 @@ public function test_sort_keys_by_callback(): void
+        $this->assertSame(
+            actual: arr()->reduce(fn ($carry, $value) => $carry . ' ' . $value, 'default'),
+            expected: 'default',
+        );
+    }
+
+    public function test_chunk(): void
+    {
+        $collection = arr([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+
+        $this->assertSame(
+            actual: $collection
+                ->chunk(2, preserveKeys: false)
+                ->map(fn ($chunk) => $chunk->toArray())
+                ->toArray(),
+            expected: [
+                [1, 2],
+                [3, 4],
+                [5, 6],
+                [7, 8],
+                [9, 10],
+            ],
+        );
+
+        $this->assertSame(
+            actual: $collection
+                ->chunk(3, preserveKeys: false)
+                ->map(fn ($chunk) => $chunk->toArray())
+                ->toArray(),
+            expected: [
+                [1, 2, 3],
+                [4, 5, 6],
+                [7, 8, 9],
+                [10],
+            ],
+        );
+    }
+
+    public function test_chunk_preserve_keys(): void
+    {
+        $collection = arr([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+
+        $this->assertSame(
+            actual: $collection
+                ->chunk(2)
+                ->map(fn ($chunk) => $chunk->toArray())
+                ->toArray(),
+            expected: [
+                [0 => 1, 1 => 2],
+                [2 => 3, 3 => 4],
+                [4 => 5, 5 => 6],
+                [6 => 7, 7 => 8],
+                [8 => 9, 9 => 10],
+            ],
+        );
+
+        $this->assertSame(
+            actual: $collection
+                ->chunk(3)
+                ->map(fn ($chunk) => $chunk->toArray())
+                ->toArray(),
+            expected: [
+                [0 => 1, 1 => 2, 2 => 3],
+                [3 => 4, 4 => 5, 5 => 6],
+                [6 => 7, 7 => 8, 8 => 9],
+                [9 => 10],
+            ],
+        );
+    }
+
+    public function test_search_with_simple_value(): void
+    {
+        $collection = arr(['apple', 'banana', 'orange']);
+
+        $this->assertSame(1, $collection->search('banana'));
+        $this->assertSame(0, $collection->search('apple'));
+        $this->assertFalse($collection->search('grape'));
+    }
+
+    public function test_search_with_strict_comparison(): void
+    {
+        $collection = arr([1, '1', 2, '2']);
+
+        // Non-strict
+        $this->assertSame(0, $collection->search(1, strict: false));
+        $this->assertSame(0, $collection->search('1', strict: false));
+
+        // Strict
+        $this->assertSame(0, $collection->search(1, strict: true));
+        $this->assertSame(1, $collection->search('1', strict: true));
+    }
+
+    public function test_search_with_closure(): void
+    {
+        $collection = arr([
+            ['id' => 1, 'name' => 'John'],
+            ['id' => 2, 'name' => 'Jane'],
+            ['id' => 3, 'name' => 'Bob'],
+        ]);
+
+        // Recherche par condition
```

Comment:
Please translate the comments in this method to English.

Reply:
Oops you're right I'm so sorry
Thanks for notice 🙏🙏🙏